tasks/request_files/request_files.py

```diff
@@ -138,19 +137,26 @@ def task(
         )
         retry_sleep_secs = DEFAULT_RESTORE_RETRY_SLEEP_SECS
 
-    # Get retrieval type
-    try:
-        retrieval_type = os.environ[OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY]
-        if retrieval_type not in ("Standard", "Bulk", "Expedited"):
-            msg = (
-                f"Invalid RESTORE_RETRIEVAL_TYPE: '{retrieval_type}'"
-                " defaulting to 'Standard'"
-            )
-            LOGGER.info(msg)
-            retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
-    except KeyError:
-        LOGGER.warn("Invalid RESTORE_RETRIEVAL_TYPE: 'None' defaulting to 'Standard'")
+    VALID_RESTORE_TYPES = ["Bulk", "Expedited", "Standard"] 
+
+    # Get config and env set to None if the key does not exist
+    env_retrieval_type = os.getenv(OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY, None)
+    config_retrieval_type = event["config"].get(CONFIG_RESTORE_DEFAULT_RETRIEVAL_TYPE_OVERRIDE_KEY, None)
+    if env_retrieval_type in VALID_RESTORE_TYPES:
+        LOGGER.info(f"Found restore type from {OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY}: {env_retrieval_type}")
+    # Set initial default to deployment env value
+    retrieval_type = env_retrieval_type
+
+    # Overide with config if we are able
+    if config_retrieval_type is not None:
+        if config_retrieval_type in VALID_RESTORE_TYPES:
```

Comment:
If config_retrieval_type is an illegal value, this will fail silently. Given the logic for going to the default below, I say remove this VALID_RESTORE_TYPES check.